File: my-folder/0001-two-sum/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int diff;
        int length=nums.length;
        int res[]=new int[2];
        boolean flag=true;
        for(int i=0;i<length&&flag;i++){
            diff=target-nums[i]; //10
            for(int j=i+1;j<length;j++){
                if(diff==nums[j]){
                    res[0]=i;
                    res[1]=j;
                    flag=false;
                    break;
                }
            }
        }

        return res;
    }
}

File: my-folder/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>res;
        if(nums.size()==0)
            return res;
        unordered_map<int,int>m;
        int diff=0;
        
        for(int i=0;i<nums.size();i++)
        {
            diff=target-nums[i];
            if(m.find(nums[i])!=m.end())
            {
                res.push_back(m[nums[i]]);
                res.push_back(i);
                return res;
            }
            else
            {
                m[diff]=i;
            }
        }
        
        return res;
        
    }
};
